src/app/clusters/basic-information/basic-information.cpp

```diff
@@ -91,228 +85,167 @@ CHIP_ERROR EncodeStringOnSuccess(CHIP_ERROR status, AttributeValueEncoder & enco
-        break;
-    }
-
-    case VendorID::Id: {
-        uint16_t vendorId = 0;
-        status            = deviceInfoProvider->GetVendorId(vendorId);
-        if (status == CHIP_NO_ERROR)
-        {
-            status = aEncoder.Encode(vendorId);
-        }
-        break;
-    }
-
-    case ProductName::Id: {
-        constexpr size_t kMaxLen      = DeviceLayer::ConfigurationManager::kMaxProductNameLength;
-        char productName[kMaxLen + 1] = { 0 };
-        status                        = deviceInfoProvider->GetProductName(productName, sizeof(productName));
-        status                        = EncodeStringOnSuccess(status, aEncoder, productName, kMaxLen);
-        break;
-    }
-
-    case ProductID::Id: {
-        uint16_t productId = 0;
-        status             = deviceInfoProvider->GetProductId(productId);
-        if (status == CHIP_NO_ERROR)
-        {
-            status = aEncoder.Encode(productId);
-        }
-        break;
-    }
-
-    case HardwareVersion::Id: {
-        uint16_t hardwareVersion = 0;
-        status                   = deviceInfoProvider->GetHardwareVersion(hardwareVersion);
-        if (status == CHIP_NO_ERROR)
-        {
-            status = aEncoder.Encode(hardwareVersion);
-        }
-        break;
-    }
-
-    case HardwareVersionString::Id: {
-        constexpr size_t kMaxLen                = DeviceLayer::ConfigurationManager::kMaxHardwareVersionStringLength;
-        char hardwareVersionString[kMaxLen + 1] = { 0 };
-        status = deviceInfoProvider->GetHardwareVersionString(hardwareVersionString, sizeof(hardwareVersionString));
-        status = EncodeStringOnSuccess(status, aEncoder, hardwareVersionString, kMaxLen);
-        break;
-    }
-
-    case SoftwareVersion::Id: {
-        uint32_t softwareVersion = 0;
-        status                   = configManager.GetSoftwareVersion(softwareVersion);
-        if (status == CHIP_NO_ERROR)
-        {
-            status = aEncoder.Encode(softwareVersion);
-        }
-        break;
-    }
+    constexpr size_t kMaxLen     = DeviceLayer::ConfigurationManager::kMaxVendorNameLength;
+    char vendorName[kMaxLen + 1] = { 0 };
+    return EncodeStringOnSuccess(deviceInfoProvider->GetVendorName(vendorName, sizeof(vendorName)), aEncoder, vendorName, kMaxLen);
+}
 
-    case SoftwareVersionString::Id: {
-        constexpr size_t kMaxLen                = DeviceLayer::ConfigurationManager::kMaxSoftwareVersionStringLength;
-        char softwareVersionString[kMaxLen + 1] = { 0 };
-        status = deviceInfoProvider->GetSoftwareVersionString(softwareVersionString, sizeof(softwareVersionString));
-        status = EncodeStringOnSuccess(status, aEncoder, softwareVersionString, kMaxLen);
-        break;
-    }
+CHIP_ERROR ReadVendorID(DeviceInstanceInfoProvider * deviceInfoProvider, AttributeValueEncoder & aEncoder)
+{
+    uint16_t vendorId = 0;
+    ReturnErrorOnFailure(deviceInfoProvider->GetVendorId(vendorId));
```

Comment:
Here you have a check during `deviceInfoProvider->Get*` but such patter is not in every function, is that intentional?

Reply:
I checked and it seems like all functions have some form of check:
  - like this one with return error on failure
  - the EncodeStringOnSuccess call which will return the failure
  - ReardManufacturingDate has its own handling which is a bit more complex
  
  Do you have a specific place you were looking at?

Reply:
Ooo my bad... Now I've looked closely and I see (:

```c++
return EncodeStringOnSuccess(deviceInfoProvider.GetVendorName ...
```

vs

```c++
ReturnErrorOnFailure(deviceInfoProvider.GetVendorId(vendorId));
return aEncoder.Encode(vendorId);
```

The different pattern has tripped me... sorry for that :/